File: Xamarin/Xamarin.Native/Xamarin.Native.iOS/Services/LocationTestService.cs
using CoreLocation;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Xamarin.iOS;
using Xamarin.Services;

namespace Xamarin.iOS
{
    public delegate void LocationChangedEventHandler(double latitude, double longitude);

    public class LocationTestService
    {
        public event LocationChangedEventHandler LocationChanged;

        CLLocationManager locationManager;

        public LocationTestService()
        {
            locationManager = new CLLocationManager();
            locationManager.LocationsUpdated += LocationsUpdated;
            locationManager.DistanceFilter = 10;
            locationManager.DesiredAccuracy = CLLocation.AccuracyBest;
        }

        public void GetLocation()
        {
            locationManager.RequestWhenInUseAuthorization();
            locationManager.StartUpdatingLocation();
        }

        public void Stop()
        {
            locationManager.StopUpdatingLocation();
        }

        private void LocationsUpdated(object sender, CLLocationsUpdatedEventArgs e)
        {
            var position = e.Locations.First().Coordinate;
            LocationChanged?.Invoke(position.Latitude, position.Longitude);
        }
    }
}

File: Native.iOS/Native.iOS/Services/LocationTestService.m
//
//  LocationTestService.m
//  Native.iOS
//
//  Created by Pawel Grzmil on 08/05/16.
//  Copyright © 2016 Pawel Grzmil. All rights reserved.
//

#import "LocationTestService.h"
#import <CoreLocation/CoreLocation.h>

@interface LocationTestService () <CLLocationManagerDelegate>

@property (nonatomic) CLLocationManager *locationManager;

@end

@implementation LocationTestService

- (instancetype)init{
    self = [super init];
    if (self) {
        self.locationManager = [[CLLocationManager alloc] init];
        self.locationManager.delegate = self;
        self.locationManager.distanceFilter = 10.f;
        self.locationManager.desiredAccuracy = kCLLocationAccuracyBest;
    }
    return self;
}

- (void)getLocation{
    [self.locationManager requestWhenInUseAuthorization];
    [self.locationManager startUpdatingLocation];
}

- (void)stop{
    [self.locationManager stopUpdatingLocation];
}

- (void)locationManager:(CLLocationManager *)manager didUpdateToLocation:(CLLocation *)newLocation fromLocation:(CLLocation *)oldLocation {
    if ([self.delegate conformsToProtocol:@protocol(LocationTestServiceDelegate)]) {
        [self.delegate locationChanged:newLocation.coordinate.latitude longitude:newLocation.coordinate.longitude];
    }
}

@end
